### PR TITLE
Signup: Steps (Domain): Improve Keyboard UX + Accessibility

### PR DESCRIPTION
**Pull request by ItsJonQ**
#### Changes + Fixes

This PR improves the overall keyboard UX and adds accessibility support for the `Popover` component that appears within the Domain Search Filter component within the sign up flow.

## Improve keyboard tab interaction for Domain Search Filter dropdown

Screen Recording 2019-05-06 at 10 24 PM

This update improves the keyboard tab interaction by prioritizing the focus
of the primary action (Apply) first, before the secondary action (Reset).

With this update, the tab order now flows like:

1. Text Field
2. Checkbox
3. Checkbox
4. Apply
5. Reset

Previously, it flowed like:

1. Text Field
2. Checkbox
3. Checkbox
4. Reset
5. Apply

Visually, the UI looks exactly the same.

The idea for this is to better align the user intention for keyboard navigation. If this interaction is desirable and something we'd want to use throughout the app, then we can create a new Component that wraps action elements (typically Buttons) to automatically handle the order. At the moment, the styles are added directly to the wrapping element.

## Popover + Search Filters Dropdown: Improving focus interaction + accessibility

Screen Recording 2019-05-06 at 10 23 PM

(GIF: Shows complete keyboard navigation with Mac OS Voiceover turned on! What you see in the darker box is what voiceover is reading to the user. It now recognizes that the Popover is a proper popover.. er.. [Tooltip](https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip))

This update enhances the base `Popover` component with improved keyboard
interactions + accessibility support.

A new Popover prop has been added - `isFocusOnShow` (`bool`). By default,
it is `false`, which ensures that the experience for current Popover use
remains the same.

However, once that prop is set to true, on show, the focus will be shifted
to the `Popover`.

Aria roles, labels, and descriptions have been added to ensure proper
screen-reader/voiceover support.

The first place that the Popover will become focused on open would be within
the Domain Search Filters Dropdown.

Aria labels and ids have been added to the trigger (context) button to
associate it with the Popover.

Since this update is on the base `Popover` component, that means it keyboard + accessibility can be improved site/app wide, if desired. Again, since I didn't want to make wide sweeping changes, the focus feature is currently opt-in.

## Button Focus style fixes

Lastly, the Button styles within the Search Filter has been adjusted to better match the hover/focus styles of the surrounding Input component.

**Before**
Screen Recording 2019-05-06 at 10 18 PM

**After**
Screen Recording 2019-05-06 at 10 21 PM

## Docs, PropTypes, and Debugs

For the `Popover` component I've added debugs (where useful), as well as `propTypes`, `defaultProps`, and some documentation to the README.md.

#### Testing instructions

1. Create a new Wordpress.com account
2. Go through the sign up flow until `/start/domains`
3. Click on the filters button, the Popover should open as normal
4. Popover should focus on open
5. Press tab to tab around
6. Turn on voiceover if you're curious, and try using the Input/Popover component

#### Review

cc'ing @lamosty and @jsnmoon for review! 🙏 (Thanks!)

#### Additional notes

I wanted to write some basic tests for the `Popover`. When I tried, I got these errors from Jest:

```
 FAIL  client/components/popover/test/popover.jsx
  ● Test suite failed to run

    ReferenceError: window is not defined

      at Object.<anonymous> (node_modules/component-event/index.js:1:39)
      at Object.<anonymous> (node_modules/click-outside/index.js:6:1)
```

After some poking around, it looks like Jest is set up with `node`, not `JSDOM`, so things like `window`, and `document` aren't available by default.

I checked some of the other components, and they didn't look like they had tests.
Hopefully this is okay for now. If not, let me know, and I'll do my best to add some. I just thought I'd get some thoughts before I go.. mocking a bunch of things from `window` 😊

### CODE REVIEW

**Comment by jsnajdr** [comment]
@shaunandrews is a frequent contributor to the Domain Search CSS styles. Flagging him for review 🙂

**Comment by ItsJonQ** [comment]
@jsnajdr Awesome! Thank you so much :D!

File: client/components/domains/search-filters/style.scss [hunk 1]
```diff
@@ -15,6 +14,9 @@
 
 	.button {
 		align-items: center;
+		border-radius: 3px;
```
Comment (shaunandrews) [inline client/components/domains/search-filters/style.scss]: Could we use the shorthand here? 0 3px 3px 0

Comment (ItsJonQ) [reply client/components/domains/search-filters/style.scss]: @shaunandrews You bet! Will make that change now <3

**Comment by ItsJonQ** [comment]
> If you'd like, I can go ahead and deploy this on Monday; let me know!

@jsnmoon That would be awesome 😍 

> If you'd like to try your hand at writing some tests for the Popover component, your effort would be greatly appreciated!

@jsnmoon Awesome! I'll look into that in the very near future :)

Thanks a bunch! 🙏

**Comment by jsnmoon** [comment]
@ItsJonQ Your changes have been deployed!

**Comment by ItsJonQ** [comment]
@jsnmoon Amazing!!! Thank you 🙏

**Pull request merged**